pkg/component/component.go

```diff
@@ -326,6 +325,31 @@ func (r *RuntimeSpecs) PolicyToComponents(policy map[string]interface{}) ([]Comp
 	return components, componentIdsInputMap, nil
 }
 
+// Injects or creates a policy.revision sub-object in the input map.
+func injectInputPolicyID(fleetPolicy map[string]interface{}, input map[string]interface{}) {
+	// If there is no top level fleet policy revision, there's nothing to inject.
+	revision, exists := fleetPolicy["revision"]
+	if !exists || input == nil {
+		return
+	}
+
+	// Check if a policy key exists with a non-nil policy object.
+	policyObj, exists := input["policy"]
+	if exists && policyObj != nil {
+		// If the policy object converts to map[string]interface{}, inject the revision key.
+		// Note that if the interface conversion here fails, we do nothing because we don't
+		// know what type of object exists with the policy key.
+		if policyMap, ok := policyObj.(map[string]interface{}); ok {
+			policyMap["revision"] = revision
+		}
+	} else {
+		// If there was no policy key or the value was nil, then inject a policy object with a revision key.
+		input["policy"] = map[string]interface{}{
+			"revision": revision,
```

Comment:
nit: linter was complaining before when "revision" was not extracted as a constant.

Reply:
I disabled that linter because it is pointless :)